types/TestResult.ts

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+// import type {ConsoleBuffer} from './Console';
+
+export type RawFileCoverage = {
+  path: string;
+  s: {[statementId: number]: number};
+  b: {[branchId: number]: number};
+  f: {[functionId: number]: number};
+  l: {[lineId: number]: number};
+  fnMap: {[functionId: number]: any};
+  statementMap: {[statementId: number]: any};
+  branchMap: {[branchId: number]: any};
+  inputSourceMap?: Record<string, any>;
+};
+
+export type RawCoverage = {
+  [filePath: string]: RawFileCoverage;
+};
+
+type FileCoverageTotal = {
+  total: number;
+  covered: number;
+  skipped: number;
+  pct: number;
+};
+
+export type CoverageSummary = {
+  lines: FileCoverageTotal;
+  statements: FileCoverageTotal;
+  branches: FileCoverageTotal;
+  functions: FileCoverageTotal;
+  merge: (other: CoverageSummary) => void;
+};
+
+export type FileCoverage = {
+  getLineCoverage: () => Record<string, any>;
+  getUncoveredLines: () => number[];
+  getBranchCoverageByLine: () => Record<string, any>;
+  toJSON: () => Record<string, any>;
+  merge: (other: Record<string, any>) => void;
+  computeSimpleTotals: (property: string) => FileCoverageTotal;
+  computeBranchTotals: () => FileCoverageTotal;
+  resetHits: () => void;
+  toSummary: () => CoverageSummary;
+};
+
+export type CoverageMap = {
+  merge: (data: Record<string, any>) => void;
+  getCoverageSummary: () => FileCoverage;
+  data: RawCoverage;
+  addFileCoverage: (fileCoverage: RawFileCoverage) => void;
+  files: () => string[];
+  fileCoverageFor: (file: string) => FileCoverage;
+};
+
+export type SerializableError = {
+  code?: unknown;
```

Comment:
Is there a reason for using `unknown` here but `any` everywhere else?

Reply:
I was following a conversion guide.  Interestingly, this type seems to exist in the [Jest repo](https://github.com/facebook/jest/blob/master/packages/jest-test-result/src/types.ts#L20-L25) and they have defined it as `unknown` also.  I'm not particularly fussed about changing it to `any`, so if you think that's better, I'll change it.

I am interested in the fact that the types are defined in Jest however...

Reply:
Since they use `unknown` I would also stick to that.